packages/aws-cdk-lib/aws-ecs/lib/container-definition.ts

```diff
@@ -590,6 +628,8 @@ export class ContainerDefinition extends Construct {
     if (props.ulimits) {
       this.addUlimits(...props.ulimits);
     }
+
+    this.validateRestartPolicy(props.restartIgnoredExitCodes, props.restartAttemptPeriod);
```

Comment:
This validation function is called in the constructor, not in the `renderRestartPolicy` function (line 1550).

```ts
function renderRestartPolicy(restartIgnoredExitCodes?: number[], restartAttemptPeriod?: cdk.Duration): CfnTaskDefinition.RestartPolicyProperty {
  return {
    enabled: true,
    ignoredExitCodes: restartIgnoredExitCodes,
    restartAttemptPeriod: restartAttemptPeriod?.toSeconds(),
  };
}
```

The render function is called in [the renderContainerDefinition](https://github.com/go-to-k/aws-cdk/blob/2c599132eab5a98ac11b13bec46f3126ba58e4e0/packages/aws-cdk-lib/aws-ecs/lib/container-definition.ts#L883) function, which [is called by Lazy() in the TaskDefinition](https://github.com/go-to-k/aws-cdk/blob/2c599132eab5a98ac11b13bec46f3126ba58e4e0/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L486). Therefore, the rendering does not run in the construction phase (even if the construct is created with `new ContainerDefinition()`, it still does not run). However, it is better to run the validation earlier, so the validation is not done in the render function, but in the constructor, which allows it to run earlier. This ensures that the validation is performed when the construct is created with `new ContainerDefinition()`.